common/src/test/groovy/com/mx/path/core/common/connect/RetryConfigurationTest.groovy

```diff
@@ -0,0 +1,165 @@
+package com.mx.path.core.common.connect
+
+import static org.mockito.Mockito.any
+import static org.mockito.Mockito.mock
+import static org.mockito.Mockito.times
+import static org.mockito.Mockito.verify
+import static org.mockito.Mockito.when
+
+import java.time.Duration
+import java.util.concurrent.Callable
+
+import com.github.rholder.retry.RetryException
+import com.mx.path.gateway.configuration.ConfigurationError
+import com.mx.path.gateway.configuration.ConfigurationState
+import com.mx.testing.WithMockery
+import com.mx.testing.connect.TestRequest
+
+import spock.lang.Specification
+
+class RetryConfigurationTest extends Specification implements WithMockery {
+
+  def subject
+  Request request
+  RequestFilter requestFilter
+
+  def setup() {
+    requestFilter = mock(RequestFilter)
+    request = new TestRequest(requestFilter)
+  }
+
+  def "fixed sleep retry"() {
+    given:
+    subject = RetryConfiguration.builder()
+        .stopStrategy(RetryConfiguration.StopStrategy.COUNT)
+        .count(2)
+        .retryOn(Collections.singletonList(ResponseMatcher.builder().predicate({ t -> true }).build()))
+        .build()
+
+    request.withRetryer(subject.instance())
+
+    when:
+    request.execute()
+
+    then:
+    verify(requestFilter, times(2)).execute(any(Request), any(Response))
+  }
+
+  def "defaults"() {
+    given:
+    subject = RetryConfiguration.defaults()
+    request.withRetryer(subject.instance())
+
+    when:
+    request.execute()
+
+    then:
+    verify(requestFilter, times(1)).execute(any(Request), any(Response))
+  }
+
+  def "verify"() {
+    given:
+    def configurationState = mock(ConfigurationState)
+
+    when:
+    RetryConfiguration.defaults().validate(configurationState)
+
+    then:
+    noExceptionThrown()
+
+    when:
```

Comment:
you could add "empty configuration" or other string descriptions after the `when:` clause to help clarify what condition is being validated

Reply:
Fixed. and cleaned up this test. It had some overlap with "validation errors".